### PR TITLE
feat: add version deps command [modversion]

### PR DESCRIPTION
**Pull request by Kubuxu**
- [x] Tests

### CODE REVIEW

File: core/commands/version.go [hunk 1]
```diff
@@ -83,3 +88,50 @@ var VersionCmd = &cmds.Command{
...
+			return
+		}
+		res.Emit(toDependency(&info.Main))
+		for _, dep := range info.Deps {
+			res.Emit(toDependency(dep))
+		}
+		return nil
+	},
+	Encoders: cmds.EncoderMap{
+		cmds.Text: cmds.MakeTypedEncoder(func(req *cmds.Request, w io.Writer, dep Dependency) error {
```
Comment (Kubuxu) [inline core/commands/version.go]: For some reason, if I change `Dependency` to `*Dependency` here I get no output, no error and the encoder is not being run (tested with `panic`).

**Comment by Kubuxu** [comment]
@Stebalien this requires us to bump to Go1.12, is that ok?

File: core/commands/version.go (hunk 1 above)
Comment (Stebalien) [reply core/commands/version.go]: https://github.com/ipfs/go-ipfs-cmds/pull/155

File: core/commands/version.go [hunk 2]
```diff
@@ -83,3 +88,50 @@ var VersionCmd = &cmds.Command{
...
+			fmt.Fprintf(w, "%s@%s", dep.Path, dep.Version)
+			if dep.ReplacedBy != "" {
+				fmt.Fprintf(w, " => %s", dep.ReplacedBy)
+			}
+			fmt.Fprintf(w, "\n")
+			return errors.New("test")
```
Comment (Stebalien) [inline core/commands/version.go]: foobar

**Comment by Stebalien** [comment]
@Kubuxu personally, I'm fine with a go 1.12 bump. Non-buggy go modules require that as well.

---

Other comment: needs test.

File: core/commands/version.go (hunk 2 above)
Comment (Kubuxu) [reply core/commands/version.go]: Seems like this error gets ignored. Interesting.

Comment (Stebalien) [reply core/commands/version.go]:
```
diff --git a/core/commands/version.go b/core/commands/version.go
index 9c5c666d0..c9632f799 100644
--- a/core/commands/version.go
+++ b/core/commands/version.go
@@ -118,9 +118,13 @@ Print out all dependencies and their versions.`,
 			}
 			return
 		}
-		res.Emit(toDependency(&info.Main))
+		if err := res.Emit(toDependency(&info.Main)); err != nil {
+			return err
+		}
 		for _, dep := range info.Deps {
-			res.Emit(toDependency(dep))
+			if err := res.Emit(toDependency(dep)); err != nil {
+				return err
+			}
 		}
 		return nil
 	},
```

**Comment by Stebalien** [comment]
@Kubuxu we'll also need to update the go version in CI.

**Comment by Kubuxu** [comment]
I know, just didn't have time to handle it today.

**Comment by Kubuxu** [comment]
This is ready.

File: test/sharness/t0010-basic-commands.sh [hunk 3]
```diff
@@ -41,6 +41,15 @@ test_expect_success "ipfs version --all has all required fields" '
   grep "Golang version" version_all.txt
 '
 
+test_expect_success "ipfs version deps succeeds" '
+  ipfs version deps >deps.txt
+'
+
+test_expect_success "ipfs version deps output looks good" '
+  head -1 deps.txt | grep "go-ipfs (devel)" &&
+  [[ $(tail -n +2 | egrep -v -c "^[^ @]+@v[^ @]+( => [^ @]+@v[^ @]+)?$") -eq 0 ]] ||
```
Comment (Stebalien) [inline test/sharness/t0010-basic-commands.sh]: Was there supposed to be another command here?

Comment (Stebalien) [reply test/sharness/t0010-basic-commands.sh]: Ah.

Comment (Kubuxu) [reply test/sharness/t0010-basic-commands.sh]: Fixing.

Comment (Kubuxu) [reply test/sharness/t0010-basic-commands.sh]: Fixed.

**Comment by Kubuxu** [comment]
Wait with it, I've done something weird in sharness.

**Comment by Kubuxu** [comment]
I think it is done.

**Comment by Kubuxu** [comment]
It is ready.

**Pull request merged**